src/analysis/mesh/qgsmeshcontours.cpp

```diff
@@ -43,13 +44,10 @@ QgsMeshContours::QgsMeshContours( QgsMeshLayer *layer )
   mNativeMesh.reset( new QgsMesh() );
   mMeshLayer->dataProvider()->populateMesh( mNativeMesh.get() );
 
-  QgsMapSettings mapSettings;
-  mapSettings.setExtent( mMeshLayer->extent() );
-  mapSettings.setDestinationCrs( mMeshLayer->crs() );
-  mapSettings.setOutputDpi( 96 );
-  QgsRenderContext context = QgsRenderContext::fromMapSettings( mapSettings );
-  mTriangularMesh.reset( new QgsTriangularMesh() );
-  mTriangularMesh->update( mNativeMesh.get(), &context );
+
+  QgsCoordinateTransform transform( mMeshLayer->crs(), mMeshLayer->crs(), QgsProject::instance() );
```

Comment:
same here -- gotta make the transform context an explicit argument in the constructor